File: ios/Plugin/AsamPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(AsamPlugin, "Asam",
    CAP_PLUGIN_METHOD(setASAM, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(enableASAM, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(disableASAM, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(isASAMEnabled, CAPPluginReturnPromise);
)

File: ios/Plugin/AsamPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AsamPlugin)
public class AsamPlugin: CAPPlugin {
    private let implementation = Asam()

    @objc func setASAM(_ call: CAPPluginCall) {
        let enable = call.getBool("enable") ?? false

        implementation.setASAM(enable, completion: { success in
            call.resolve([
                "success": success
            ])
        })
    }

    @objc func enableASAM(_ call: CAPPluginCall) {
        implementation.setASAM(true, completion: { success in
            call.resolve([
                "success": success
            ])
        })
    }

    @objc func disableASAM(_ call: CAPPluginCall) {
        implementation.setASAM(false, completion: { success in
            call.resolve([
                "success": success
            ])
        })
    }

    @objc func isASAMEnabled(_ call: CAPPluginCall) {
        implementation.isASAMEnabled { enabled in
            call.resolve([
                "enabled": enabled
            ])
        }
    }

    @objc func isSupervised(_ call: CAPPluginCall) {
        implementation.isSupervised { supervised in
            call.resolve([
                "supervised": supervised
            ])
        }
    }
}
